File: LHDS.Core/Models/Foundations/TerminologyArtifacts/TerminologyArtifact.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

using System;
using LHDS.Core.Models.Bases;

namespace LHDS.Core.Models.Foundations.TerminologyArtifacts
{
    public class TerminologyArtifact : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string FullUrl { get; set; }
        public string ResourceType { get; set; }
        public string? Version { get; set; }
        public string? Name { get; set; }
        public string? Title { get; set; }
        public string? Status { get; set; }
        public DateTimeOffset? LastUpdated { get; set; }
        public bool IsCore { get; set; }
        public bool IsDownloaded { get; set; }
        public bool IsForUser { get; set; }
        public bool IsDownloadedForUser { get; set; }
        public bool IsError { get; set; }
        public string? ErrorMessage { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public string UpdatedBy { get; set; } = string.Empty;
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
    }
}

File: LHDS.AdminPortal.Configurations/lhds.adminportal.configurations.client/src/models/terminologyArtifacts/terminologyArtifact.ts

export class TerminologyArtifact {
    public id: string;
    public fullUrl?: string;
    public resourceType?: string;
    public version?: string;
    public name?: string;
    public title?: string;
    public status?: string;
    public lastUpdated?: Date;
    public isCore?: boolean;
    public isDownloaded?: boolean;
    public isError?: boolean;
    public errorMessage?: string;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;

    constructor(terminologyArtifact: TerminologyArtifact) {
        this.id = terminologyArtifact.id ? terminologyArtifact.id : "";
        this.fullUrl = terminologyArtifact.fullUrl;
        this.resourceType = terminologyArtifact.resourceType;
        this.version = terminologyArtifact.version;
        this.name = terminologyArtifact.name;
        this.title = terminologyArtifact.title;
        this.status = terminologyArtifact.status;
        this.lastUpdated = new Date(terminologyArtifact.lastUpdated);
        this.isCore = terminologyArtifact.isCore;
        this.isDownloaded = terminologyArtifact.isDownloaded;
        this.isError = terminologyArtifact.isError;
        this.errorMessage = terminologyArtifact.errorMessage;
        this.createdBy = terminologyArtifact.createdBy;
        this.createdDate = new Date(terminologyArtifact.createdDate);
        this.updatedBy = terminologyArtifact.updatedBy;
        this.updatedDate = new Date(terminologyArtifact.updatedDate);
    }
}